File: LeetCode/1984. Minimum Difference Between Highest and Lowest of K Scores/solution.cpp
/**
 * 118 / 118 test cases passed.
 * Runtime: 16 ms
 * Memory Usage: 13.3 MB 
 */
class Solution {
public:
    int minimumDifference(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());
        int lowest = nums[0], highest = nums[k - 1];
        int ans = highest - lowest;
        for (int i = k; i < nums.size(); ++ i) {
            highest = nums[i];
            lowest = nums[i - k + 1];
            ans = min(ans, highest - lowest);
        }
        return ans;
    }
};

File: LeetCode/1984. Minimum Difference Between Highest and Lowest of K Scores/solution.py
"""
118 / 118 test cases passed.
Runtime: 52 ms
Memory Usage: 15.1 MB
"""
class Solution:
    def minimumDifference(self, nums: List[int], k: int) -> int:
        nums.sort()
        lowest, highest = nums[0], nums[k - 1]
        ans = highest - lowest
        for i in range(k, len(nums)):
            lowest, highest = nums[i - k + 1], nums[i]
            ans = min(ans, highest - lowest)
        return ans
